direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/Utils.java

```diff
@@ -19,36 +19,55 @@
 import cz.o2.proxima.storage.UriUtil;
 import java.net.URI;
 import java.util.Collection;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
 import org.apache.kafka.clients.consumer.KafkaConsumer;
-import org.apache.kafka.common.TopicPartition;
 
 /** Various utilities. */
 class Utils {
 
+  static final String TOPIC_PATTERN_QUERY = "topicPattern";
+
   /**
    * Retrieve topic from given URI.
    *
    * @param uri the URL
    * @return topic name
    */
+  @Nullable
```

Comment:
Maybe Optional<String> would be better return type for this method.

Reply:
That is internal API, I like better to use Optionals on public methods only.